persist/sqlite/consensus_refactored_test.go

```diff
@@ -570,6 +600,78 @@ func TestV2TransactionChainIndices(t *testing.T) {
 	n.assertV2ChainIndices(t, txn2.ID())
 }
 
+func TestMaturedSiacoinBalance(t *testing.T) {
+	pk1 := types.GeneratePrivateKey()
+	uc1 := types.StandardUnlockConditions(pk1.PublicKey())
+	addr1 := uc1.UnlockHash()
+
+	n := newTestChain(t, false, nil)
+
+	checkBalance := func(addr types.Address, expectedSC, expectedImmatureSC types.Currency) {
+		t.Helper()
+
+		sc, immatureSC, sf, err := n.db.Balance(addr)
+		if err != nil {
+			t.Fatal(err)
+		}
+		testutil.Equal(t, "siacoins", expectedSC, sc)
+		testutil.Equal(t, "immature siacoins", expectedImmatureSC, immatureSC)
+		testutil.Equal(t, "siafunds", 0, sf)
+	}
+
+	checkBalance(addr1, types.ZeroCurrency, types.ZeroCurrency)
+
+	b := testutil.MineBlock(n.tipState(), nil, addr1)
+	n.applyBlock(t, b)
+
+	val := b.MinerPayouts[0].Value
+
+	for range n.network.MaturityDelay {
+		checkBalance(addr1, types.ZeroCurrency, val)
+		n.mineTransactions(t)
+	}
+
+	checkBalance(addr1, val, types.ZeroCurrency)
+
+	for range n.network.MaturityDelay {
+		n.revertBlock(t)
+		checkBalance(addr1, types.ZeroCurrency, val)
+	}
```

Comment:
we can add a final check `checkBalance(addr1, types.ZeroCurrency, types.ZeroCurrency)`